### PR TITLE
Added Filtering for tags

### PR DESCRIPTION
**Pull request by vpramo**
Added Filtering for tags, for now it supports single filters
in the format below - snapshots/?tag=searchtag

### CODE REVIEW

File: aasemble/django/apps/api/v1/views.py [hunk 1]
```diff
@@ -105,9 +105,15 @@ class SnapshotViewSet(viewsets.ModelViewSet):
             serializer_class = selff.serializers.SnapshotSerializer
 
             def get_queryset(self):
-                if self.request.user.is_superuser:
-                    return self.queryset.all()
-                return self.queryset.filter(mirrorset__owner_id=self.request.user.id)
+                tag=self.request.query_params.get('tag', None)
+                if tag is not None:
+                    if self.request.user.is_superuser:
+                        return self.queryset.all(tags__tag=tag)
```
Comment (sorenisanerd) [inline aasemble/django/apps/api/v1/views.py]: I'm fairly sure this is not valid. Does .all() even take any arguments, ever?

File: aasemble/django/apps/api/v1/views.py [hunk 2]
```diff
@@ -105,9 +105,15 @@ class SnapshotViewSet(viewsets.ModelViewSet):
...
+                    return self.queryset.filter(mirrorset__owner_id=self.request.user.id, tags__tag=tag)
+                else:
+                    if self.request.user.is_superuser:
+                        return self.queryset.all()
+                    return self.queryset.filter(mirrorset__owner_id=self.request.user.id)
```
Comment (sorenisanerd) [inline aasemble/django/apps/api/v1/views.py]: I would flip this whole thing upside down. Start by constructing a queryset based on whether self.request.user is superuser or not:

```
if self.request.user.is_superuser:
    qs = self.queryset.all()
else:
    qs = self.queryset.filter(<filter by user id here>)

tag = self.request.query_params.get('tag', None)
if tag is not None:
    qs.filter(<whatever you need to do to filter by the tag>)

return qs
```

File: aasemble/django/apps/buildsvc/test_data/debian/build/debian/changelog [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+
+ -- aaSemble Package Builder <pkgbuild@aasemble.com>  Fri, 27 Nov 2015 12:07:56 +0000
+
+buildsvctest (0.1+10) aasemble; urgency=medium
+
+  * Automatically built.
+    SHA: 
+
+ -- aaSemble Package Builder <pkgbuild@aasemble.com>  Fri, 27 Nov 2015 11:28:15 +0000
+
```
Comment (sorenisanerd) [inline aasemble/django/apps/buildsvc/test_data/debian/build/debian/changelog]: Can you revert these changes, please?

**Comment by vpramo** [comment]
@soren its done

**Comment by sorenisanerd** [comment]
Great work!
